R/exposure_metrics.R

```diff
@@ -132,3 +132,71 @@ calc_auc_from_regimen <- function(
     target_design$type
   )
 }
+
+#' Calculate time to target attainment
+#'
+#' @param regimen PKPDsim regimen object
+#' @param target_design target design, created using `create_target_design()`
+#' @param auc_comp compartment to look for AUC
+#' @param model PKPDsim model object
+#' @param terminal_interval duration of final dosing interval
+#' @param ... arguments passed on to `PKPDsim::sim`
+#' @return a numeric value indicating the time, in hours, if a dosing interval
+#'   contained the target metric within range. Otherwise, returns NA.
+#' @export
+calc_time_to_target <- function(
+    regimen,
+    target_design,
+    auc_comp,
+    model,
+    terminal_interval = 24,
+    ...
+) {
+  target_type <- match.arg(tolower(target_design$type), mipd_target_types())
+
+  iov <- PKPDsim::get_model_iov(model)
+  if (is.null(iov[["bins"]])) iov[["bins"]] <- c(0, 9999)
```

Comment:
is this intended to be passed to PKPDsim::sim a couple lines below? if so, this step was omitted.